Always use self.appbuilder.get_session in security managers

uranusjr opened the pull request:
Currently there's inconsistent usages of self.get_session and self.appbuilder.get_session. These are the same thing, except in ApplessAirflowSecurityManager (where appbuilder isn't available).

I don't think this inconsistency currently causes any bugs, but it has potential to be problematic down the road.

hussein-awala reviewed (commented):
I think creating a FakeAppBuilder is more clean :thinking: 
```python
class FakeAppBuilder(AppBuilder):
    def __int__(self, session: Session | None = None):
        self.session = session


class ApplessAirflowSecurityManager(AirflowSecurityManager):
    """Security Manager that doesn't need the whole flask app"""

    def __init__(self, session: Session | None = None):
        self.appbuilder = FakeAppBuilder(session)
```

uranusjr committed:
Unify how AirflowSecurityManager accesses db

Instead of having two distinct ways to get a database session, one only
for upstream calls and one only for the "appless" subclass, this
patch introduces a "fake" appbuilder attribute that allows us to unify
the call pattern, preventing potential bugs caused by using the wrong
call for the wrong class.

hussein-awala reviewed (approved):
Nice! LGTM
